### DISCUSSION TITLE
Cara mencari mean untuk tipe data <dbl>

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#534] [Q&A] [answered]
Saya lagi mencari rata-rata untuk daily return saham. Contoh untuk saham AALI, saya mencari daily returnnya dengan cara sebagai berikut:

`AALI <- AALI[order(AALI$Date,decreasing =T),]
AALI$Adj.Close <- NULL
AALI$Daily_Return <- NULL
AALI$dret <- c(-diff(AALI$Close)/AALI$Close[-1]*100,NA)`

Hasilnya adalah sebagai berikut:
![image](https://user-images.githubusercontent.com/122977451/217026880-5ba0851f-e758-49d8-b61e-f2e902fcb6a6.png)

Bisa dilihat bahwa tipe data untuk dret adalah dbl. Awalnya saya ubah saja dengan fungsi as.numeric. Namun masalah muncul ketika ingin mencari rata-rata untuk kolom tersebut. Ketika saya mencari rata-rata dengan fungsi mean(), hasilnya justru NA, seperti gambar berikut:
![image](https://user-images.githubusercontent.com/122977451/217028578-6731fe8f-3dff-4abb-b34f-6d1470636b6f.png)

Apakah ada cara perubahan yang perlu lakukan untuk tipe data dbl?

**Comment by kevbow** [comment, marked as answer]
Dear @AGelfP,

Tipe data dbl dapat langsung dihitung nilai mean-nya tanpa diubah tipenya karena merupakan penyebutan data numeric pada dataframe.
Anda mendapatkan hasil NA karena pada data Anda terdapat nilai NA. Hal ini dapat diselesaikan dengan menambahkan parameter na.rm = T pada fungsi mean(). Silakan gunakan kode berikut:
```r
mean(AALI$dret, na.rm = T)
```
Demikian yang dapat saya sampaikan, semoga membantu.

**Reply by AGelfP** [reply]
ok, sudah bisa Kak. Terima kasih yaa
